### ISSUE TITLE
MacOS Install with remotes fails looking for libproj

### ISSUE THREAD

**Issue by dblodgett-usgs** [r-spatial/sf#1312]
Sorry to open yet another issue on this.

I am running up to date gdal (3.0.4_2) and proj (7.0.0) from osgeo4mac.

Working through some dplyr and sf issues in a package and need to figure out how to get the latest working locally before the CRAN release.

I have pkg-config and have uninstalled all competing Brew kegs. Am I missing something?

<details><summary>Install output</summary>

```r
> remotes::install_github("r-spatial/sf")
Downloading GitHub repo r-spatial/sf@master
✓  checking for file ‘/private/var/folders/p6/kr6wq6bj7wvdz1qywzhksz5h001t8d/T/RtmpstZDYP/remotesa6fc35ce9516/r-spatial-sf-66e51d8/DESCRIPTION’ ...
─  preparing ‘sf’:
✓  checking DESCRIPTION meta-information ...
─  cleaning src
─  running ‘cleanup’
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  building ‘sf_0.9-0.tar.gz’
   Warning: invalid uid value replaced by that for user 'nobody'
   Warning: invalid gid value replaced by that for user 'nobody'
   
* installing *source* package ‘sf’ ...
** using staged installation
configure: CC: clang
configure: CXX: clang++ -std=gnu++11
checking for gdal-config... /usr/local/bin/gdal-config
checking gdal-config usability... yes
configure: GDAL: 3.0.4
checking GDAL version >= 2.0.1... yes
checking for gcc... clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether clang accepts -g... yes
checking for clang option to accept ISO C89... none needed
checking how to run the C preprocessor... clang -E
checking for grep that handles long lines and -e... /usr/bin/grep
checking for egrep... /usr/bin/grep -E
checking for ANSI C header files... rm: conftest.dSYM: is a directory
rm: conftest.dSYM: is a directory
yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking gdal.h usability... yes
checking gdal.h presence... yes
checking for gdal.h... yes
checking GDAL: linking with --libs only... yes
checking GDAL: /usr/local/Cellar/osgeo-gdal/3.0.4_2/share/gdal/pcs.csv readable... no
checking GDAL: checking whether PROJ is available for linking:... yes
checking GDAL: checking whether PROJ is available fur running:... yes
configure: GDAL: 3.0.4
checking proj.h usability... yes
checking proj.h presence... yes
checking for proj.h... yes
checking PROJ: checking whether PROJ and sqlite3 are available for linking:... no
configure: error: libproj not found in standard or given locations.
ERROR: configuration failed for package ‘sf’
* removing ‘/Library/Frameworks/R.framework/Versions/3.6/Resources/library/sf’
Error: Failed to install 'sf' from GitHub:
  (converted from warning) installation of package ‘/var/folders/p6/kr6wq6bj7wvdz1qywzhksz5h001t8d/T//RtmpstZDYP/filea6fcd9df2f3/sf_0.9-0.tar.gz’ had non-zero exit status
```

</details>

**Comment by edzer** [comment]
Newer PROJ needs sqlite3, to read it's proj.db database. The error suggests problems finding it.

**Comment by dblodgett-usgs** [comment]
OK -- Do you know if anyone is running the latest sf on Mac with gdal3/proj7? Curious if there's a secret sauce to success. The primary brew gdal is on 2.4.4 still.

**Comment by dblodgett-usgs** [comment]
OK -- I wiped out my brew and re-upped from scratch with `brew install osgeo-gdal`

I can get past the error above with the addition of configure.args:
```
remotes::install_github("r-spatial/sf", configure.args = "--with-gdal-config=/usr/local/bin/gdal-config --with-geos-config=/usr/local/bin/geos-config --with-proj-include=/usr/local/include/ --with-proj-lib=/usr/local/lib/")
```
(taken from: https://github.com/r-spatial/sf/issues/1092)

But then I get errors that appear to be the same as: https://github.com/RcppCore/Rcpp/pull/1047

I installed `Rcpp` from github and bob's your uncle.

**Comment by dblodgett-usgs** [comment]
Note for future me googling the error message, I ran into issues with rgdal over riding the proj.db path or something... 

error:
`GDAL Error 1: PROJ: proj_create_from_database: Cannot find proj.db`

If I install rgdal with:
```
install.packages("rgdal", type = "source", configure.args = "--with-gdal-config=/usr/local/bin/gdal-config --with-geos-config=/usr/local/bin/geos-config --with-proj-include=/usr/local/include/ --with-proj-lib=/usr/local/lib/")
```
Things work.

**Comment by luispuerto** [comment]
Hey! 

Thanks a lot, I was having also problem installing and this solved the issue. 

Question, is this because there is something not correctly configured in the installation of the brew formula from osgeo tap or what is going on here?

**Comment by DanOvando** [comment]
HUGE thanks, been banging my head against this for the last 24 hours. Any idea why this suddenly became an issue? I had the same problem on two different mac laptops running Catalina, and have been installing `sf` and `gdal` via homebrew with absolutely no problems for a long time. Glad to have a fix but would like to understand why it broke in the first place

> OK -- I wiped out my brew and re-upped from scratch with `brew install osgeo-gdal`
> 
> I can get past the error above with the addition of configure.args:
> 
> ```
> remotes::install_github("r-spatial/sf", configure.args = "--with-gdal-config=/usr/local/bin/gdal-config --with-geos-config=/usr/local/bin/geos-config --with-proj-include=/usr/local/include/ --with-proj-lib=/usr/local/lib/")
> ```
> 
> (taken from: #1092)
> 
> But then I get errors that appear to be the same as: [RcppCore/Rcpp#1047](https://github.com/RcppCore/Rcpp/pull/1047)
> 
> I installed `Rcpp` from github and bob's your uncle.

**Comment by dblodgett-usgs** [comment]
Far as I can tell it's something to do with using the new proj but I'm really not sure. All my googling seemed to end with the `LIB_PROJ` environment variable and how and where it is set was and is still a total mystery to me.

**Comment by DanOvando** [comment]
Weird, well thanks for finding a solution. @edzer is it worth adding this to the the install instructions in the README for `sf`?  Seems like a few people are having the same problem

**Comment by edzer** [comment]
Well, it seems a Rcpp problem really, no? If you think it is worth it, please submit a PR.

**Comment by ateucher** [comment]
It does seem like something has changed in installing on macOS, and I'm not sure if it's to do with `sf` or the configuration of `proj` installed by homebrew. Regardless, I do need to set `configure.args` to install from source now whereas I did not before. After experimentation I did get it down to:

```r
install.packages("sf", configure.args = "--with-proj-lib=/usr/local/lib/")
```

I also learned that you can set package-specific `configure.args` via options, so now I have set in my `.Rprofile` file:

```r
options(configure.args = c("sf" = '--with-proj-lib=/usr/local/lib/'))
```

So I can install `sf` from source with just `install.packages("sf")` and it reads that option.

NB: I tried setting the `PROJ_LIB` env var to `/usr/local/lib/` thinking this would have the same effect as setting the `--with-proj-lib` configure.arg but it did not.

I'd be happy to submit a PR to update the README if you'd like @edzer.

**Comment by edzer** [comment]
Yes, great idea, @ateucher !

**Comment by rsbivand** [comment]
@ateucher 

> I tried setting the PROJ_LIB env var to /usr/local/lib/ thinking this would have the same effect as setting the --with-proj-lib configure.arg but it did not.

`PROJ_LIB` is the environment variable pointing to where the PRØJ metadata files are stored, in your case `/usr/local/share/proj`. Despite the name, it is not where `libproj` lives.

**Comment by ateucher** [comment]
Thanks @rsbivand, that makes sense... I'm slowly figuring these things out. Is the environment variable used at build time or run time (or both?)

**Comment by ateucher** [comment]
I also found that to install `rgdal` from source (which I needed to do to make it play nice with `sf 0.9`, I needed the additional flag `--with-proj-include=/usr/local/include/`. Having env vars `PROJ_LIB`, `PKG_CONFIG_PATH`, and `GDAL_DATA` set or unset seems to make no difference.

**Comment by rsbivand** [comment]
The `PROJ_LIB` environmental variable only applies to run time, never build etc. It may be read when programs build using PRØJ to locate the metadata files.

I suspect that the underlying  problems are associated with a slightly defective `proj.pc` file used by `pkg-config` in `./configure` if available.

**Comment by edzer** [comment]
... which, afaict, should be resolved in proj 7.0.1.
